src/GNDStk/Node/src/all.hpp

```diff
@@ -0,0 +1,35 @@
+
+// -----------------------------------------------------------------------------
+// Node::all(string)
+//
+// Searches the node's children for all child nodes of the given name. Returns
+// a container of copies of all such child nodes. (Might be size 0, of course.)
+// -----------------------------------------------------------------------------
+
+// const
+template<template<class...> class CONTAINER = std::vector>
+CONTAINER<Node,std::allocator<Node>> all(
+   const std::string &key,
+   bool &found = detail::default_bool
+) const {
+   // container
+   CONTAINER<Node,std::allocator<Node>> container;
+
+   try {
+      // search
+      for (auto &c : children)
+         if (c->name == key)
+            container.push_back(c->copy()); // can throw
```

Comment:
Re: copy() - see my other new comment.
Re: ranges. Yes, let's look into this sometime. For now, let's heed the advice of either Donald Knuth or C.A.R. Hoare (each of whom credits the other): "We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil. Yet we should not pass up our opportunities in that critical 3%."

Reply:
Note of course that in this code, there's both the filling of the container, and the container return from the function. The latter, at least, would involve move construction or assignment, so not a big deal. Whether or not the node copies (into the container) are expensive, or not expensive, depends a great deal on what's in the nodes, and where they are in the overall tree. Note also that this function deal with generic node access; with luck, the whole meta_t/child_t "smart keyword" system will be much preferred by users, so that they're getting back real objects of some reasonable type - which would need to be built on-the-fly anyway.